### PR TITLE
3 - Add Test AccountDeletionTest

### PR DESCRIPTION
**Pull request by florencialecha**
This pull request includes the addition of a new test, "shouldDeleteAllAccountsWhenUserClickToDeleteData", under the scope of AccountDeletionTest.

This test is designed to ensure that when a user chooses to delete all data, all associated accounts are successfully deleted. 

Note: The functionality introduced in this update plays a crucial role in verifying that the system behaves correctly when a user opts to delete all data within the 'Account Deletion' feature of our application. It's also important to mention that 'deleteData' is the final step in all tests related to the 'View Checking' feature, ensuring the clean-up of data after each test run.

Additionally:
We have rectified the 'SingleAccountActivationTest' class by adding a teardown function.

### CODE REVIEW

File: qa-automation/src/test/java/com/digitalbank/qaautomation/tests/myProfile/AccountDeletionTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Description("When a user chooses to delete all data, all accounts should be deleted.")
+    @Test
+    public void shouldDeleteAllAccountsWhenUserClickToDeleteData() {
+        driver.get("http://digitalbank.upcamp.io/bank/login");
+        LoginPage loginPage = new LoginPage(driver);
+
+        // Log in with a user
+        SideBarPage sideBarPage = loginPage.logIn("jsmith@demo.io", "Demo123!");
+
+        //El login me debería llevar a una home que tiene una sidebar y una header page
```
Comment (JISaldana) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/tests/myProfile/AccountDeletionTest.java]: translate this so all comments are in the same language

File: qa-automation/src/test/java/com/digitalbank/qaautomation/tests/viewChecking/AccountExistViewTest.java [hunk 2]
```diff
@@ -1,2 +1,59 @@
...
+    @Feature("Checking Account Display")
+    @Story("Display account cards and transaction history when checking accounts exist")
+    @Description("When a user with checking accounts views their checking accounts, account cards and transaction history should be displayed.")
+    @Test
+    public void shouldDisplayCheckingViewPageWhenAccountsExist() {
+
+        driver.get("http://digitalbank.upcamp.io/bank/login");
+        LoginPage loginPage = new LoginPage(driver);
+
+        //Pasarle usuario con cuentas de checking existentes
```
Comment (JISaldana) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/tests/viewChecking/AccountExistViewTest.java]: translate this too

**Review by JISaldana — DISMISSED** [review]

File: qa-automation/src/test/java/com/digitalbank/qaautomation/tests/myProfile/AccountDeletionTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        HeaderPage headerPage = new HeaderPage(driver);
+        headerPage.clickUserProfileBtn();
+        headerPage.clickDeleteDataButton();
+
+        // Check redirection to home
```
Comment (sofiambd) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/tests/myProfile/AccountDeletionTest.java]: Tratar de no usar comentarios, con Steps en pages es suficientes.

File: qa-automation/src/test/java/com/digitalbank/qaautomation/tests/myProfile/AccountDeletionTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        String currentUrl = driver.getCurrentUrl();
+        Assert.assertEquals(currentUrl, "http://digitalbank.upcamp.io/bank/home");
```
Comment (sofiambd) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/tests/myProfile/AccountDeletionTest.java]: es buena practica Strings definirlas y usarlas en codigo, no "quemar" el texto

File: qa-automation/src/test/java/com/digitalbank/qaautomation/tests/newChecking/NewCheckingAccountCreation.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Story("Creation of different types of checking accounts")
+    @Description("When a user enters valid data in the fields, a new checking account should be created.")
+    @Test(dataProvider = "accountData", dataProviderClass = AccountData.class)
+    public void shouldCreateNewAccountWhenValidDataIsProvided(String accountType, String ownershipType, String accountName, String initialDeposit) {
+
+        driver.get("http://digitalbank.upcamp.io/bank/login");
+        LoginPage loginPage = new LoginPage(driver);
+
+        // Log in with a user
```
Comment (sofiambd) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/tests/newChecking/NewCheckingAccountCreation.java]: Revisar comentarios primer test

File: qa-automation/src/test/java/com/digitalbank/qaautomation/tests/viewChecking/AccountExistViewTest.java (hunk 2 above)
Comment (sofiambd) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/tests/viewChecking/AccountExistViewTest.java]: Revisar comentarios primer test

File: qa-automation/src/test/java/com/digitalbank/qaautomation/tests/viewChecking/SingleAccountActivationTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        checkingViewPage.activateAccount(1);
+        boolean isFirstAccountActive = checkingViewPage.isAccountActive(1);
+        Assert.assertTrue(isFirstAccountActive, "The first account should be active after activation");
+
+        // Activate the second account
+        checkingViewPage.activateAccount(2);
+        boolean isSecondAccountActive = checkingViewPage.isAccountActive(2);
+        Assert.assertTrue(isSecondAccountActive, "The second account should be active after activation");
+
+        // Verify that the first account is deactivated
```
Comment (sofiambd) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/tests/viewChecking/SingleAccountActivationTest.java]: Revisar comentarios primer test.

File: qa-automation/src/test/java/com/digitalbank/qaautomation/pages/HeaderPage.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+    public HeaderPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public void clickUserProfileBtn() {
+        userProfileBtn.click();
+    }
+
+    public HomePage clickDeleteDataButton() {
```
Comment (sofiambd) [inline qa-automation/src/test/java/com/digitalbank/qaautomation/pages/HeaderPage.java]: Faltan Steps

**Review by sofiambd — CHANGES_REQUESTED** [review]
Tratar de mantener los PR lo más pequeños posibles, como dijimos en la working session de Buenas practicas, deberia ser 1 PR por test.

**Pull request closed without merging**